packages/core/test/authorization.test.ts

```diff
@@ -0,0 +1,217 @@
+import { test, expect } from 'vitest'
+import { execute, parse } from 'graphql'
+
+let counter = 0
+const importFuse = () => {
+  return import('../src/builder?test-authorization=' + counter++)
+}
+
+test('Should correctly do top-level field authorization', async () => {
+  const mod = await importFuse()
+  const { builder, addQueryFields, defineAuthScopes } = mod
+  defineAuthScopes((ctx) => ({ isLoggedIn: !!ctx.user }))
+
+  addQueryFields((t) => ({
+    isLoggedIn: t.field({
+      type: 'Boolean',
+      description: 'An authentication check',
+      authScopes: {
+        isLoggedIn: true,
+      },
+      resolve: () => {
+        return true
+      },
+    }),
+  }))
+
+  const schema = builder.toSchema()
+
+  const document = parse(`query { isLoggedIn }`)
+  let result = (await execute({
+    document,
+    schema,
+    contextValue: {},
+  })) as { data?: any; errors?: any }
+
+  expect(result.data).toEqual({ isLoggedIn: null })
+  expect(result.errors).toBeDefined()
+  expect(result.errors).toHaveLength(1)
+  expect(result.errors[0].message).toEqual('Not authorized')
```

Comment:
Ahh another reason I don't like the implementation of `Query.isLoggedIn` it returns null for false and puts the false in an error. I don't consider that query to have errored yet it returns an error.

Reply:
This is just a test 😅